src/link/abstractlink.h

```diff
@@ -212,10 +212,24 @@ class AbstractLink : public QObject
     void write(const char* data, int size)
     {
         if(size > 0) {
-            emit sendData(QByteArray(data, size));
-        };
+            if (_type == LinkType::Serial) {
+                writeSync(data, size);
```

Comment:
why don't you make this the default behavior of the serial port ?
If this is link specific should be in the link and not in abstract link.
If you are trying to avoid the eventloop signal-slot handle, you just need to make the connect attribute as direct and not queue.

Reply:
> why don't you make this the default behavior of the serial port ?
If this is link specific should be in the link and not in abstract link.

I put this case here because I have not tested other link types. I think that maybe we should favor the writeSync logic over the emit sendData() logic for everyone.

> If you are trying to avoid the eventloop signal-slot handle, you just need to make the connect attribute as direct and not queue.

what magic do you speak of?